### PR TITLE
[ios][android] Add wheelchair info to place page

### PR DESCRIPTION
**Pull request by dvdmrtnz**
This PR adds wheelchair info to the place page allowing to distinguish between `wheelchair=no`, `wheelchair=limited` and `wheelchair=yes`.

https://wiki.openstreetmap.org/wiki/Key:wheelchair

### CODE REVIEW

**Review by vng — COMMENTED** [review]
LGTM. Android will be later or you don't want to dive into it?

**Comment by dvdmrtnz** [comment]
@vng I have not added it to Android because I am not sure how to access the `GetLocalizedWheelchairType()` function from Android. If you have some suggestions on how to handle it I can give it a go. Otherwise this can be added just to iOS at the moment.

**Comment by dvdmrtnz** [comment]
@vng I feel like a refactoring here could help with adding these attributes that are derived from the classificator types. Maybe they could be generated in core c++ code and then passed as an array of tags, a bit like you said in PR https://github.com/organicmaps/organicmaps/pull/7108#issuecomment-1890429469.

**Comment by vng** [comment]
Well, definitely we can implement an array with a predefined params structure to add similar-classifier attributes here.
You can try or I can try a bit later. At this moment your solution is good and acceptable.

File: indexer/feature_utils.cpp [hunk 1]
```diff
@@ -438,6 +438,16 @@ string GetLocalizedFeeType(TypesHolder const & types)
   return localized_types[0];
 }
 
+string GetLocalizedWheelchairType(TypesHolder const & types)
+{
+  auto const & isWheelchairType = ftypes::IsWheelchairTypeChecker::Instance();
+  auto localized_types = GetLocalizedTypes(isWheelchairType, types);
```
Comment (biodranik) [inline indexer/feature_utils.cpp:444]:
```suggestion
  auto localizedTypes = GetLocalizedTypes(isWheelchairType, types);
```

Comment (biodranik) [reply indexer/feature_utils.cpp:444]: Underscores are used only in member prefixed variables starting from m_

File: iphone/CoreApi/CoreApi/PlacePageData/Common/PlacePageInfoData.mm [hunk 2]
```diff
@@ -83,6 +83,9 @@ - (instancetype)initWithRawData:(Info const &)rawData ohLocalization:(id<IOpenin
     
     _atm = rawData.HasAtm() ? NSLocalizedString(@"type.amenity.atm", nil) : nil;
       
+    std::string wheelchair = rawData.GetLocalizedWheelchairType();
+    _wheelchair = !wheelchair.empty() ? ToNSString(wheelchair) : nil;
```
Comment (biodranik) [inline iphone/CoreApi/CoreApi/PlacePageData/Common/PlacePageInfoData.mm:87]: Isn't _wheelchair always nil by default?
```suggestion
    if (!wheelchair.empty())
      _wheelchair = @(wheelchair.c_str());
```

File: iphone/CoreApi/CoreApi/PlacePageData/Common/PlacePageInfoData.mm [hunk 3]
```diff
@@ -83,6 +83,9 @@ - (instancetype)initWithRawData:(Info const &)rawData ohLocalization:(id<IOpenin
     
     _atm = rawData.HasAtm() ? NSLocalizedString(@"type.amenity.atm", nil) : nil;
       
+    std::string wheelchair = rawData.GetLocalizedWheelchairType();
```
Comment (biodranik) [inline iphone/CoreApi/CoreApi/PlacePageData/Common/PlacePageInfoData.mm:86]: How editing of wheelchair accessibility will be supported? Usually, when we add some fields into Place Page, we assume that they are/will be editable.

Does it make sense to introduce and pass an enum from the C++ core, and localize it in the UI? Or just some hard-coded keys-strings "unknown", "yes", "no", and "limited"?

Also think about Android implementation to make it easier. Maybe something similar to already existing editor implementations like WiFi editing?

**Comment by dvdmrtnz** [comment]
I have changed how this works. I have added a new metadata property that gets filled at runtime. This way I can add this to android the same way as the rest of the properties.

Also the metadata uses the readable wheelchair type and gets localized in the UI.

File: android/app/src/main/java/app/organicmaps/bookmarks/data/Metadata.java [hunk 4]
```diff
@@ -60,7 +60,8 @@ public enum MetadataType
     FMD_JUNCTION_REF(39),
     FMD_BUILDING_MIN_LEVEL(40),
     FMD_WIKIMEDIA_COMMONS(41),
-    FMD_CAPACITY(42);
+    FMD_CAPACITY(42),
+    FMD_WHEELCHAIR(43);
```
Comment (vng) [inline android/app/src/main/java/app/organicmaps/bookmarks/data/Metadata.java:64]: This approach is ok I think. Just to add a comment here: The value is runtime only, but actual data is taken from the classifier types.

I suppose we will remove it from the classifier someday and add it to the metadata.

**Review by vng — APPROVED** [review]

**Comment by vng** [comment]
@rtsisyk Why do we have _legal_ group in reviewers here? :)

**Comment by Jean-BaptisteC** [comment]
> @rtsisyk Why do we have _legal_ group in reviewers here? :)

Because this PR update files in assets directory

File: indexer/feature_meta.hpp [hunk 5]
```diff
@@ -148,6 +148,7 @@ class Metadata : public MetadataBase
     FMD_BUILDING_MIN_LEVEL = 40,
     FMD_WIKIMEDIA_COMMONS = 41,
     FMD_CAPACITY = 42,
+    FMD_WHEELCHAIR = 43,
```
[the remark this answers is not in the document]
Comment (dvdmrtnz) [reply indexer/feature_meta.hpp]: I added a comment

**Pull request merged**